models/dodo/ethereum/dodo_ethereum_trades.sql

```diff
@@ -0,0 +1,348 @@
+        WHERE evt_block_time >= date_trunc("day", now() - interval '1 week')
+        {% endif %}
+        WHERE trader <> '0xa356867fdcea8e71aeaf87805808803806231fdc'
+    ) dexs
+    INNER JOIN {{ source('ethereum', 'transactions')}} tx
+        ON dexs.tx_hash = tx.hash
+         {% if not is_incremental() %}
+        AND tx.block_time >= '{{project_start_date}}'
+        {% endif %}
+        AND tx.block_time < CURRENT_TIMESTAMP()
+        {% if is_incremental() %}
+        AND tx.block_time >= date_trunc("day", now() - interval '1 week')
+        {% endif %}
+        AND tx.block_number >={{query_start_block_number}}
+        AND tx.block_number < {{query_end_block_number}}
+    LEFT JOIN {{ ref('tokens_erc20') }} erc20a ON erc20a.contract_address = dexs.token_a_address
+    LEFT JOIN {{ ref('tokens_erc20') }} erc20b ON erc20b.contract_address = dexs.token_b_address
+    LEFT JOIN {{ source('prices', 'usd')}} pa ON pa.minute = date_trunc('minute', dexs.block_time)
+        AND pa.contract_address = dexs.token_a_address
+        {% if not is_incremental() %}
+        AND pa.minute >= '{{project_start_date}}'
+        {% endif %}
+        {% if is_incremental() %}
+        AND pa.minute >= date_trunc("day", now() - interval '1 week')
+        {% endif %}
+        AND pa.minute < CURRENT_TIMESTAMP()
+    LEFT JOIN {{ source('prices', 'usd')}} pb ON pb.minute = date_trunc('minute', dexs.block_time)
+        AND pb.contract_address = dexs.token_b_address
+        {% if not is_incremental() %}
+        AND pb.minute >= '{{project_start_date}}'
+        {% endif %}
+        {% if is_incremental() %}
+        AND pb.minute >= date_trunc("day", now() - interval '1 week')
+        {% endif %}
+        AND pb.minute < CURRENT_TIMESTAMP()
+        --FIGURING OUT A WAY TO INCLUDE THE prices.layer1_usd 
+    -- LEFT JOIN prices.layer1_usd pe ON pe.minute = date_trunc('minute', dexs.block_time)
+    --     AND pe.symbol = 'ETH'
+    --     AND pe.minute >= '2020-08-10 13:19'
+    --     AND pe.minute < CURRENT_TIMESTAMP()
+    WHERE dexs.block_time >= '{{project_start_date}}'
+    AND dexs.block_time < CURRENT_TIMESTAMP()
+    AND dexs.token_a_address <> dexs.token_b_address
+  )
+
+SELECT
+    'ethereum' AS blockchain,
+    project,
+    version,
+    category,
+    TRY_CAST(date_trunc('DAY', dexs.block_time) AS date) AS block_date,
+    dexs.block_time,
+    token_a_symbol AS token_bought_symbol,
+    token_b_symbol AS token_sold_symbol,
+    case
+        when lower(token_a_symbol) > lower(token_b_symbol) then concat(token_b_symbol, '-', token_a_symbol)
+        else concat(token_a_symbol, '-', token_b_symbol)
+    end AS token_pair,
+    token_a_amount AS token_bought_amount,
+    token_b_amount AS token_sold_amount,
+    token_a_amount_raw AS token_bought_amount_raw,
+    token_b_amount_raw AS token_sold_amount_raw,
+    usd_amount AS amount_usd,
+    token_a_address AS token_bought_address,
+    token_a_address AS token_sold_address,
```

Comment:
Here I made a mistake too.
It should be `token_b_address AS token_sold_address,`

Reply:
good catch -- exactly why we apply the generic test with values in the seed!

Reply:
Thanks  alot @Hosuke 

@jeff-dude  forgive my lack of proactivity. I was out just got back home. Let me do the changes.